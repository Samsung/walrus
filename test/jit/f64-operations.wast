(module
(func (export "test11") (result f64)
    f64.const 3.3
    f64.const 4.4
    f64.max
)
(func (export "test12") (result f64)
    f64.const 3.3
    f64.const 4.4
    f64.min
)
(func (export "test13") (result f64)
    f64.const 4.0
    f64.sqrt
)
(func (export "test14") (result f64)
    f64.const 3.3
    f64.ceil
)
(func (export "test15") (result f64)
    f64.const 3.3
    f64.floor
)
(func (export "test16") (result f64)
    f64.const 3.3
    f64.trunc
)
(func (export "test17") (result f64)
    f64.const 3.3
    f64.nearest
)
(func (export "test18") (result f64)
    f64.const 3.3
    f64.abs
)
(func (export "test19") (result f64)
    f64.const -3.3
    f64.abs
)
(func (export "test20") (result f64)
    f64.const -3.3
    f64.neg
)
(func (export "test21") (result f64)
    f64.const -3.3
    f64.const 3.3
    f64.copysign
)
)

(assert_return (invoke "test11") (f64.const 4.4))
(assert_return (invoke "test12") (f64.const 3.3))
(assert_return (invoke "test13") (f64.const 2.0))
(assert_return (invoke "test14") (f64.const 4.0))
(assert_return (invoke "test15") (f64.const 3.0))
(assert_return (invoke "test16") (f64.const 3.0))
(assert_return (invoke "test17") (f64.const 3.0))
(assert_return (invoke "test18") (f64.const 3.3))
(assert_return (invoke "test19") (f64.const 3.3))
(assert_return (invoke "test20") (f64.const 3.3))
(assert_return (invoke "test21") (f64.const 3.3))
