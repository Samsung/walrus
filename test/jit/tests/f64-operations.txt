(module
(func (export "test11") (result f64)
    f64.const 3.3
    f64.const 4.4
    f64.max
)
(func (export "test12") (result f64)
    f64.const 3.3
    f64.const 4.4
    f64.min
)
(func (export "test13") (result f64)
    f64.const 4.0
    f64.sqrt
)
(func (export "test14") (result f64)
    f64.const 3.3
    f64.ceil
)
(func (export "test15") (result f64)
    f64.const 3.3
    f64.floor
)
(func (export "test16") (result f64)
    f64.const 3.3
    f64.trunc
)
(func (export "test17") (result f64)
    f64.const 3.3
    f64.nearest
)
(func (export "test18") (result f64)
    f64.const 3.3
    f64.abs
)
(func (export "test19") (result f64)
    f64.const -3.3
    f64.abs
)
(func (export "test20") (result f64)
    f64.const -3.3
    f64.neg
)
(func (export "test21") (result f64)
    f64.const -3.3
    f64.const 3.3
    f64.copysign
)
)
