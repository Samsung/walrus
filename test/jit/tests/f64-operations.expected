[[[[[[[  Function   0  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f32.const
  Result[0]: f64
2: Opcode: f64.max
  Param[0]: ref:0(0)
  Param[1]: ref:1(0)
  Result[0]: f64
3: Opcode: return
  Param[0]: ref:2(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f32.const
  Result[0]: f64 unused
2: Opcode: f64.max
  Param[0]: f64 const:0
  Param[1]: f64 const:1
  Result[0]: f64 sp[0]
3: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   1  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f32.const
  Result[0]: f64
2: Opcode: f64.min
  Param[0]: ref:0(0)
  Param[1]: ref:1(0)
  Result[0]: f64
3: Opcode: return
  Param[0]: ref:2(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f32.const
  Result[0]: f64 unused
2: Opcode: f64.min
  Param[0]: f64 const:0
  Param[1]: f64 const:1
  Result[0]: f64 sp[0]
3: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   2  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.sqrt
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.sqrt
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   3  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.ceil
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.ceil
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   4  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.floor
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.floor
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   5  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.trunc
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.trunc
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   6  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.nearest
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.nearest
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   7  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.abs
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.abs
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   8  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.abs
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.abs
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function   9  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f64.neg
  Param[0]: ref:0(0)
  Result[0]: f64
2: Opcode: return
  Param[0]: ref:1(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f64.neg
  Param[0]: f64 const:0
  Result[0]: f64 sp[0]
2: Opcode: return
  Param[0]: f64 sp[0]

[[[[[[[  Function  10  ]]]]]]]
Number of locals reduced from 0 to 0
0: Opcode: f32.const
  Result[0]: f64
1: Opcode: f32.const
  Result[0]: f64
2: Opcode: f64.copysign
  Param[0]: ref:0(0)
  Param[1]: ref:1(0)
  Result[0]: f64
3: Opcode: return
  Param[0]: ref:2(0)
------------------------------
Frame size: 0, Arguments size: 16 (total: 16)
0: Opcode: f32.const
  Result[0]: f64 unused
1: Opcode: f32.const
  Result[0]: f64 unused
2: Opcode: f64.copysign
  Param[0]: f64 const:0
  Param[1]: f64 const:1
  Result[0]: f64 sp[0]
3: Opcode: return
  Param[0]: f64 sp[0]

test11() => f64:4.400000
test12() => f64:3.300000
test13() => f64:2.000000
test14() => f64:4.000000
test15() => f64:3.000000
test16() => f64:3.000000
test17() => f64:3.000000
test18() => f64:3.300000
test19() => f64:3.300000
test20() => f64:3.300000
test21() => f64:3.300000
